apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryJob.cs

```diff
@@ -201,6 +189,22 @@ public BigQueryJob ThrowOnAnyError()
         public Task<BigQueryJob> CancelAsync(CancelJobOptions options = null, CancellationToken cancellationToken = default(CancellationToken)) =>
             _client.CancelJobAsync(Reference, options, cancellationToken);
 
+        private TableReference GetQueryTableReference()
+        {
+            if (Resource?.Configuration?.Query == null)
+            {
+                throw new InvalidOperationException("Job doesn't represent a query");
+            }
+            ThrowOnAnyError();
+            var reference = Resource?.Configuration?.Query?.DestinationTable;
```

Comment:
Nit: since we already know that `Resource?.Configuration?.Query` is non-null, this can be `Resource.Configuration.Query.DestinationTable`. Although I think it would be better to cache `Resource?.Configuration?.Query` is a local above, check for null, and then use it here.